src/Config/Converters/Utf8JsonReaderExtensions.cs

```diff
@@ -41,4 +42,115 @@ static internal class Utf8JsonReaderExtensions
 
         return JsonSerializer.Deserialize<string>(ref reader, options);
     }
+
+    /// <summary>
+    /// Replaces placeholders for environment variables in the given JSON string and returns the new JSON string.
+    /// </summary>
+    /// <param name="inputJson">The JSON string to process.</param>
+    /// <exception cref="JsonException">Thrown when the input is null or when an error occurs while processing the JSON.</exception>
+    public static string? ReplaceEnvVarsInJson(string? inputJson)
```

Comment:
why do we need this function when `StringJsonConverterFactory()` already exists which has helpers to parse out envvironment variable placeholders? that class also uses regex to detect instances of `@env()`

why not something like:

```csharp
TryParseConfig(
                json: json,
                config: out config,
                connectionString: _connectionString,
                replaceEnvVar: true)
```

Reply:
Is there something about `dab validate` that requires a completely different code path + new code?
if so , please add to pr description and code comments.

Reply:
please elaborate in PR description. this doesn't really tell us much about "why" it can't be used.

> There were certain methods which was doing replacement, but it was written in a way to align with the RuntimeConfig Object. So, it couldn't be used.

Reply:
> Also, we need the original conversion mechanism to stay since it will be utilized for the workflow outside of dab validate.

Not clear why original conversion would need to change.